swagger/body-array.json

```diff
@@ -70,6 +70,34 @@
         }
       }
     },
+    "/array/query": {
```

Comment:
I will admit that I have not paid enough attention since we started to semi-support openapi3 - but do we need a corresponding openapi3 test case?

Reply:
I haven't heard of special tests for open api 3 @fearthecowboy @daviwil do you guys know? if not, should we start having separate tests?

Reply:
For example, the way you describe collection formats is different between OAS2 and 3.

Reply:
It might be good to gradually start adding OpenAPI 3 test cases because more and more service teams are starting to use it.  I'm not sure the best way to integrate it into the existing testserver design, maybe @lmazuel would have some ideas.

Reply:
There is already query array tests in url.json , example `queries_ArrayStringCsvEmpty` why is this PR different?

Reply:
After reviewing url.json, I understand that this PR is adding a test without `collectionFormat`, which we don't have indeed.
@daviwil @johanste should m4 put "csv" as default, instead of asking all of us to implement it?

Reply:
About OAI3, I'm not sure yet
https://swagger.io/docs/specification/serialization/

@daviwil would you have the matrix of mapping between OA2 and OA3 in the autorest converter?

Reply:
Do you really want m4 to express it in OAS2/swagger lingo? `collectionFormat` does not exist in OAS3.

Reply:
Default in OA2: /users?id=3,4,5
Default in OA3: /users?id=3&id=4&id=5

So I still think m4 should convert the default OA2 in explicit "form/false"

Reply:
Two bugs:
- https://github.com/Azure/autorest.modelerfour/issues/315
- https://github.com/Azure/autorest.modelerfour/issues/314

Reply:
Thanks for pointing out the current issues in AutoRest v3 and Modelerfour, @lmazuel filed a couple of issues that should be pretty easy to take care of to get our conversion handling everything properly:

- https://github.com/Azure/autorest.modelerfour/issues/314
- https://github.com/Azure/perks/issues/117